livy/src/main/java/org/apache/zeppelin/livy/BaseLivyInterpreter.java

```diff
@@ -653,18 +655,24 @@ public Principal getUserPrincipal() {
     }
 
 
+    
+    RestTemplate restTemplate = null;
     if (isSpnegoEnabled) {
       if (httpClient == null) {
-        return new KerberosRestTemplate(keytabLocation, principal);
+        restTemplate = new KerberosRestTemplate(keytabLocation, principal);
       } else {
-        return new KerberosRestTemplate(keytabLocation, principal, httpClient);
+        restTemplate = new KerberosRestTemplate(keytabLocation, principal, httpClient);
       }
     }
     if (httpClient == null) {
-      return new RestTemplate();
+      restTemplate = new RestTemplate();
     } else {
-      return new RestTemplate(new HttpComponentsClientHttpRequestFactory(httpClient));
+      restTemplate = new RestTemplate(new HttpComponentsClientHttpRequestFactory(httpClient));
     }
+    if (null != restTemplate) {
```

Comment:
What case would restTemplate be null ?

Reply:
it cannot be null in any scenario , but i made a null check because the variable restTemplate is defined as null and i dont want a nullpointerException to be thrown in this part of code anytime near future .